src/main/java/com/salesforce/storm/spout/sideline/persistence/ZookeeperPersistenceAdapter.java

```diff
@@ -177,15 +178,15 @@ public SidelinePayload retrieveSidelineRequest(SidelineRequestIdentifier id, int
 
         final FilterChainStep step = parseJsonToFilterChainSteps(json);
 
-        final Long startingOffset = (Long) json.get("startingOffset");
-        final Long endingOffset = (Long) json.get("endingOffset");
+        final Double startingOffset = (Double) json.get("startingOffset");
```

Comment:
This feels weird because we use long's everywhere else.  I believe Long is the preferred way to store large integers, where Double can allow you to store LARGER numbers but with less precision (seems bad)

Reply:
Is this an artifact of using Gson?  Perhaps configuring GSON with:

```java
Gson gson = new GsonBuilder().
        registerTypeAdapter(Double.class,  new JsonSerializer<Double>() {   
       .....implementation returns a Long here...
});
```

Tho even that feels like a hack.

Reply:
This is an artifact of switching to Gson and serializing to a Map. I'd like to create a concrete class for this payload, but I think I'd prefer to do that in a separate PR if that's agreeable.

Reply:
Create an issue for it so we can capture it and I'll +1 this.

Reply:
See #54